its/src/test/java/com/sonar/it/scanner/msbuild/ScannerMSBuildTest.java

```diff
@@ -453,6 +453,18 @@ public void testCSharpSharedFiles() throws IOException {
       .isNotNull();
   }
 
+  @Test
+  public void testCSharpSharedProjectType() throws IOException {
+    runBeginBuildAndEndForStandardProject("CSharpSharedProjectType");
+
+    assertThat(getComponent("CSharpSharedProjectType:SharedProject/TestEventInvoke.cs"))
+      .isNotNull();
+    assertThat(getComponent("CSharpSharedProjectType:CSharpSharedProjectType:36F96F66-8136-46C0-B83B-EFAE05A8FFC1:Program.cs"))
+      .isNotNull();
+    assertThat(getComponent("CSharpSharedProjectType:CSharpSharedProjectType:F96D8AA1-BCE1-4655-8D65-08F2A5FAC15B:Program.cs"))
```

Comment:
Is it possible to assert that the shared file is explicitly not added to the "normal" projects?

Reply:
Not sure because calling `getComponent("invalid")` throws web exception with 404 message.